packages/vertexai/src/googleAIMappers.ts

```diff
@@ -0,0 +1,235 @@
+ *
+ * @param candidates The {@link GoogleAIGenerateContentCandidate} to map.
+ * @returns A {@link GenerateContentCandidate} that conforms to the Vertex AI format.
+ *
+ * @throws If any {@link Part} in the candidates has a `videoMetadata` property.
+ *
+ * @internal
+ */
+export function mapGenerateContentCandidates(
+  candidates: GoogleAIGenerateContentCandidate[]
+): GenerateContentCandidate[] {
+  const mappedCandidates: GenerateContentCandidate[] = [];
+  let mappedSafetyRatings: SafetyRating[];
+  if (mappedCandidates) {
+    candidates.forEach(candidate => {
+      // Map citationSources to citations.
+      let citationMetadata: CitationMetadata | undefined;
+      if (candidate.citationMetadata) {
+        citationMetadata = {
+          citations: candidate.citationMetadata.citationSources
+        };
+      }
+
+      // Assign missing candidate SafetyRatings properties to their defaults.
+      if (candidate.safetyRatings) {
+        logger.warn(
+          "Candidate safety rating properties 'severity', 'severityScore', and 'probabilityScore' are not included in responses from Google AI. Properties have been assigned to default values."
+        );
+        mappedSafetyRatings = candidate.safetyRatings.map(safetyRating => {
+          return {
+            ...safetyRating,
+            severity: HarmSeverity.HARM_SEVERITY_UNSUPPORTED,
+            probabilityScore: 0,
+            severityScore: 0
+          };
+        });
+      }
+
+      // videoMetadata is not supported.
+      // Throw early since developers may send a long video as input and only expect to pay
+      // for inference on a small portion of the video.
+      if (
+        candidate.content?.parts.some(
+          part => (part as InlineDataPart)?.videoMetadata
+        )
+      ) {
+        throw new GenAIError(
+          GenAIErrorCode.UNSUPPORTED,
+          'Part.videoMetadata is not supported in Google AI. Please remove this property.'
+        );
+      }
+
+      const mappedCandidate = {
+        index: candidate.index,
+        content: candidate.content,
+        finishReason: candidate.finishReason,
+        finishMessage: candidate.finishMessage,
+        safetyRatings: mappedSafetyRatings,
+        citationMetadata,
+        groundingMetadata: candidate.groundingMetadata
+      };
+      mappedCandidates.push(mappedCandidate);
+    });
+  }
+
+  return mappedCandidates;
+}
+
+export function mapPromptFeedback(
+  promptFeedback: PromptFeedback
+): PromptFeedback {
+  // Assign missing PromptFeedback SafetyRatings properties to their defaults.
+  const mappedSafetyRatings: SafetyRating[] = [];
+  promptFeedback.safetyRatings.forEach(safetyRating => {
+    mappedSafetyRatings.push({
+      category: safetyRating.category,
+      probability: safetyRating.probability,
+      severity: HarmSeverity.HARM_SEVERITY_UNSUPPORTED,
+      probabilityScore: 0,
+      severityScore: 0,
+      blocked: safetyRating.blocked
+    });
+  });
+  logger.warn(
```

Comment:
Is there a right way to do things where the developer can avoid this warning? I don't think developers like warnings. Maybe they should only be warned if they put something in the fields and it needs to be replaced. Otherwise filling in defaults when they put nothing probably should count as expected behavior.

Reply:
Yeah I agree warnings could be annoying. This is something Andrew and I aligned on for all SDKs, so I'll ask him about this approach of only warning if modifying user defined data

Reply:
Discussed with Andrew and decided to remove this warning.